pkg/scheduler/actions/enqueue/enqueue_test.go

```diff
@@ -0,0 +1,151 @@
+			},
+			Queues: []*schedulingv1.Queue{
+				util.BuildQueue("c1", 1, api.BuildResourceList("4", "4G")),
+				util.BuildQueue("c2", 1, api.BuildResourceList("4", "4G")),
+			},
+			Status: map[api.JobID]scheduling.PodGroupPhase{
+				"c1/pg1": scheduling.PodGroupInqueue,
+				"c1/pg2": scheduling.PodGroupInqueue,
+			},
+		},
+		{
+			Name: "when podgroup status is running",
+			PodGroups: []*schedulingv1.PodGroup{
+				util.BuildPodGroup("pg1", "c1", "c1", 0, nil, schedulingv1.PodGroupRunning),
+			},
+			Pods: []*v1.Pod{
+				util.BuildPod("c1", "p1", "", v1.PodRunning, api.BuildResourceList("1", "1G"), "pg1", make(map[string]string), make(map[string]string)),
+				util.BuildPod("c1", "p2", "", v1.PodRunning, api.BuildResourceList("1", "1G"), "pg1", make(map[string]string), make(map[string]string)),
+			},
+			Queues: []*schedulingv1.Queue{
+				util.BuildQueue("c1", 1, api.BuildResourceList("4", "4G")),
+			},
+			Status: map[api.JobID]scheduling.PodGroupPhase{
+				"c1/pg1": scheduling.PodGroupRunning,
+			},
+		},
+		{
+			Name: "pggroup cannot enqueue because the specified queue is c1, but there is only c2",
+			PodGroups: []*schedulingv1.PodGroup{
+				util.BuildPodGroup("pg1", "c1", "c1", 0, nil, schedulingv1.PodGroupPending),
+			},
+			Queues: []*schedulingv1.Queue{
+				util.BuildQueue("c2", 1, api.BuildResourceList("4", "4G")),
+			},
+			Status: map[api.JobID]scheduling.PodGroupPhase{},
+		},
+		{
+			Name: "pggroup cannot enqueue because queue resources are less than podgroup MinResources",
+			PodGroups: []*schedulingv1.PodGroup{
+				util.BuildPodGroupWithMinResources("pg1", "c1", "c1", 0,
+					nil, api.BuildResourceList("8", "8G"), schedulingv1.PodGroupPending),
+			},
+			Queues: []*schedulingv1.Queue{
+				util.BuildQueue("c1", 1, api.BuildResourceList("1", "1G")),
+			},
+			Status: map[api.JobID]scheduling.PodGroupPhase{
+				"c1/pg1": scheduling.PodGroupPending,
+			},
+		},
+	}
+
+	trueValue := true
+	tiers := []conf.Tier{
+		{
+			Plugins: []conf.PluginOption{
+				{
+					Name:               drf.PluginName,
+					EnabledJobOrder:    &trueValue,
+					EnabledJobEnqueued: &trueValue,
+				},
+				{
+					Name:               proportion.PluginName,
+					EnabledQueueOrder:  &trueValue,
+					EnabledJobEnqueued: &trueValue,
+				},
+				{
+					Name: sla.PluginName,
+					Arguments: map[string]interface{}{
+						"sla-waiting-time": "3m",
+					},
+					EnabledJobOrder:    &trueValue,
+					EnabledJobEnqueued: &trueValue,
+				},
+				{
+					Name:            gang.PluginName,
+					EnabledJobOrder: &trueValue,
+				},
+			},
+		},
+	}
+	for i, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			test.Plugins = plugins
+			test.RegisterSession(tiers, nil)
+			defer test.Close()
+			action := New()
+			test.Run([]framework.Action{action})
+			if err := test.CheckAll(i); err != nil {
+				t.Fatalf("case: (index: %v/name: %s), action: %s, error: %s", i, test.Name, action.Name(), err)
```

Comment:
duplicate error message。 just need print the error which has included detail.

Reply:
done